shank-macro-impl/src/types/resolve_rust_ty.rs

```diff
@@ -2,9 +2,8 @@ use std::{convert::TryFrom, ops::Deref};
 
```

Comment:
I'm not sure about this file, looks like all format changes?
Maybe your IDE isn't respecting the `rustfmt.toml` in the package folder?
I just ran `cargo fmt` inside that package and don't get any changes.

Reply:
Ran `cargo fmt` and fixed the formatting issues. Sorry about that 😅

Change to support `COption` was that I just made it parse as a normal `Option` in `resolve_rust_ty.rs`
 ```rust
 "COption" | "Option" => {
 ```

Reply:
Yeah I do wanna support all option types as well, but need to make sure first that they actually are de/serialized the same by borsh.
Need to add a [test case here](https://github.com/metaplex-foundation/beet/blob/master/tools/bsamples/src/options.rs).